polyfill/lib/ecmascript.mjs

```diff
@@ -161,13 +161,17 @@ export const ES = ObjectAssign({}, ES2019, {
   },
   ParseTemporalTimeZoneString: (stringIdent) => {
     try {
-      const canonicalIdent = ES.GetCanonicalTimeZoneIdentifier(stringIdent);
-      if (canonicalIdent) return { zone: canonicalIdent.toString() };
+      try {
+        const canonicalIdent = ES.GetCanonicalTimeZoneIdentifier(stringIdent);
+        if (canonicalIdent) return { zone: canonicalIdent.toString() };
+      } catch {
+        // fall through
+      }
+      // Try parsing ISO string instead
+      return ES.ParseISODateTime(stringIdent, { zoneRequired: true });
     } catch {
-      // fall through
+      throw new RangeError(`Invalid time zone: ${stringIdent}`);
```

Comment:
I added a nested exception handler here because the error message returned by the fallen-through code (`invalid ISO 8601 string: ${isoString}`) was confusing in the most common case where the user is trying to parse a time zone identifier, not a full ISO 8601 string.